src/vl-map-action.js

```diff
@@ -80,30 +117,38 @@ export class VlMapAction extends vlElement(HTMLElement) {
    */
   actionChanged() {
     const event = new Event(VlMapAction.NEW_ACTION_EVENT_NAME);
-    this.parentElement.dispatchEvent(event);
+    this._mapElement.dispatchEvent(event);
   }
 
   _layerChangedCallback() {
-    this._computeAction(this._map, this.layer);
+    this._mapElement.ready.then(() => {
+      if (this.layer) {
+        const action = this._createAction(this.layer);
+        if (action) {
+          this._mapElement.addAction(action);
+          this.actionChanged();
+          if (this._active) {
+            action.activate();
+          }
+          this._action = action;
+        }
+      }
+    });
   }
 
   _createAction() {
-    console.log('implementatie van de _createAction ontbreekt');
-  }
-
-  _computeAction(map, kaartlaag) {
-    let action;
-    if (map && kaartlaag) {
-      action = this._createAction(kaartlaag);
-      this.parentElement.addAction(action);
-      this.actionChanged();
-    }
-    this._action = action;
+    console.log('implementatie van _createAction ontbreekt');
```

Comment:
Nitpick, maar console.warn is misschien toepasselijker, mogelijks zelfs een error throwen

Reply:
Ik denk dat dit niet meer relevant is voor deze PR, dus markeer het als resolved. Anders geef je maar een seintje.